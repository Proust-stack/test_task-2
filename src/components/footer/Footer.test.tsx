import { cleanup, screen } from '@testing-library/react';
import { renderFooterWithRouter } from '../../tests/helpers/renderWithRouter';

describe('Footer component', () => {
  afterEach(cleanup);
  it('Footer snapshot', () => {
    const view = renderFooterWithRouter()
    expect(view).toMatchSnapshot();
  });

  it('items field exist and quantity equals 3', () => {
    renderFooterWithRouter();
    expect(screen.getByText(/3 items left/i)).toBeInTheDocument();
  });

  it('buttons exist', () => {
    renderFooterWithRouter();
    const btns = screen.getAllByRole('button');
    btns.forEach(btn => {
      expect(btn).toBeInTheDocument();
    });
    expect(btns).toMatchSnapshot();
  });
});
