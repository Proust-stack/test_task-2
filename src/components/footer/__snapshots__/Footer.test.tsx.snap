// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Footer snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer_container"
      >
        <div
          class="footer_items"
        >
          3
           items left
        </div>
        <div
          class="footer_filter filter"
        >
          <a
            aria-current="page"
            class="selected"
            data-testid="all-link"
            href="/"
          >
            <button
              class="filter_all"
              data-testid="all_btn"
            >
              All
            </button>
          </a>
          <a
            data-testid="active-link"
            href="/active"
          >
            <button
              class="filter_active"
              data-testid="active_btn"
            >
              Active
            </button>
          </a>
          <a
            data-testid="completed-link"
            href="/completed"
          >
            <button
              class="filter_completed"
              data-testid="completed_btn"
            >
              Completed
            </button>
          </a>
        </div>
        <button
          class="footer_clear"
          data-testid="clear_btn"
        >
          Clear completed
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer_container"
    >
      <div
        class="footer_items"
      >
        3
         items left
      </div>
      <div
        class="footer_filter filter"
      >
        <a
          aria-current="page"
          class="selected"
          data-testid="all-link"
          href="/"
        >
          <button
            class="filter_all"
            data-testid="all_btn"
          >
            All
          </button>
        </a>
        <a
          data-testid="active-link"
          href="/active"
        >
          <button
            class="filter_active"
            data-testid="active_btn"
          >
            Active
          </button>
        </a>
        <a
          data-testid="completed-link"
          href="/completed"
        >
          <button
            class="filter_completed"
            data-testid="completed_btn"
          >
            Completed
          </button>
        </a>
      </div>
      <button
        class="footer_clear"
        data-testid="clear_btn"
      >
        Clear completed
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Footer component buttons exist 1`] = `
Array [
  <button
    class="filter_all"
    data-testid="all_btn"
  >
    All
  </button>,
  <button
    class="filter_active"
    data-testid="active_btn"
  >
    Active
  </button>,
  <button
    class="filter_completed"
    data-testid="completed_btn"
  >
    Completed
  </button>,
  <button
    class="footer_clear"
    data-testid="clear_btn"
  >
    Clear completed
  </button>,
]
`;
